A baseline domain has a wildcard DNS setup leading to inaccurate results

## Description
`telegram.com` is one of the domains used by dnsvalidator to check for DNS poisoning. If a resolver responds with a non-`NXDOMAIN` response for a random subdomain, it's dismissed.
```python
nxdomainchecks = ["facebook.com", "paypal.com", "google.com",
                  "bet365.com", "telegram.com", "wikileaks.com"]
...
    for nxdomaincheck in nxdomainchecks:
        # make sure random subdomains are NXDOMAIN
        try:
            positivehn = "{rand}.{domain}".format(
                rand=rand(),
                domain=nxdomaincheck
            )
            posanswer = resolver.query(positivehn, 'A')

            # nxdomain exception was not thrown, we got records when we shouldn't have.
            # Skip the server.
            output.terminal(Level.ERROR, server,
                            "DNS poisoning detected, passing")
            return
```

The issue is that `telegram.com` is a wildcard.
```
$ dig '*.telegram.com'

; <<>> DiG 9.10.6 <<>> *.telegram.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 25491
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1220
;; QUESTION SECTION:
;*.telegram.com.			IN	A

;; ANSWER SECTION:
*.telegram.com.		9600	IN	CNAME	wh158-gbahn.production.gannettdigital.com.
wh158-gbahn.production.gannettdigital.com. 300 IN A 159.54.247.168

;; Query time: 127 msec
;; SERVER: 192.168.1.1#53(192.168.1.1)
;; WHEN: Wed Nov 02 22:36:46 EET 2022
;; MSG SIZE  rcvd: 111
```
So any resolver that correctly responds with `NOERROR` for `randomstring.telegram.com` is disregarded with a `DNS poisoning detected, passing` message.

---

## PoC
```
$ dnsvalidator -t 8.8.8.8
=======================================================
dnsvalidator v0.1	by James McLean (@vortexau)
                	& Michael Skelton (@codingo_)
=======================================================
[22:41:46] [INFO] [1.1.1.1] resolving baseline
[22:41:46] [INFO] [8.8.8.8] resolving baseline
[22:41:47] [INFO] [9.9.9.9] resolving baseline
[22:41:47] [INFO] [8.8.8.8] Checking...
[22:41:48] [ERROR] [8.8.8.8] DNS poisoning detected, passing
[22:41:48] [INFO] Finished. Discovered 0 servers
```

--- a/dnsvalidator/dnsvalidator.py
+++ b/dnsvalidator/dnsvalidator.py
@@ -34,7 +34,7 @@ baselines = ["1.1.1.1", "8.8.8.8", "9.9.9.9"]
 
 positivebaselines = ["bet365.com", "telegram.com"]
 nxdomainchecks = ["facebook.com", "paypal.com", "google.com",
-                  "bet365.com", "telegram.com", "wikileaks.com"]
+                  "bet365.com", "wikileaks.com"]
 
 goodip = ""
 valid_servers = []

